[ros2] ‘placeholders’ is not a namespace-name - missing include

ros2 branch, using gcc-7.1.1 (ArchLinux):
```
.../class_loader/include/class_loader/class_loader.h:128:26: error: ‘placeholders’ is not a namespace-name
     using namespace std::placeholders;
```
in `include/class_loader/class_loader.h` there's a missing include, adding it fixed the compilation error:
```
#include <functional>
```

It's possible that in previous versions of gcc `functional` was included in other libraries used here, but now it's not the case.

--- a/include/class_loader/class_loader.h
+++ b/include/class_loader/class_loader.h
@@ -32,6 +32,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <functional>
 #include <memory>
 #include <mutex>
 #include <string>

